crates/volta-migrate/src/v2.rs

```diff
@@ -94,24 +94,31 @@ impl TryFrom<V1> for V2 {
 /// This will ensure that we don't treat the default npm from a prior version of Volta as a "custom" npm that
 /// the user explicitly requested
 fn clear_default_npm(platform_file: &Path) -> Fallible<()> {
-    if platform_file.exists() {
-        let platform_json =
-            read_to_string(platform_file).with_context(|| ErrorKind::ReadPlatformError {
-                file: platform_file.to_owned(),
-            })?;
-        let mut existing_platform = Platform::from_json(platform_json)?;
-
-        if let Some(ref mut node_version) = &mut existing_platform.node {
-            if let Some(npm) = &node_version.npm {
-                if let Ok(default_npm) = load_default_npm_version(&node_version.runtime) {
-                    if *npm == default_npm {
-                        node_version.npm = None;
-                        write(platform_file, existing_platform.into_json()?).with_context(
-                            || ErrorKind::WritePlatformError {
-                                file: platform_file.to_owned(),
-                            },
-                        )?;
-                    }
+    let platform_json =
+        match read_to_string(platform_file).with_context(|| ErrorKind::ReadPlatformError {
+            file: platform_file.to_owned(),
+        }) {
+            Err(error) => {
+                if error.is_not_found_error_kind() {
+                    return Ok(());
+                } else {
+                    return Err(error);
+                }
+            }
+            Ok(json) => json,
+        };
```

Comment:
I think this can be simplified a little by removing the call to `with_context` and instead making it part of the error branch:

```rust
let platform_json = match read_to_string(platform_file) {
    Err(error) => {
        if error.kind() == io::ErrorKind::NotFound {
            return Ok(());
        } else {
            return Err(VoltaError::from_source(error, ErrorKind::ReadPlatformError { file: platform_file.to_owned() }));
        }
    }
    Ok(json) => json,
}
```

Reply:
Simplified by using `.or_else(ok_if_not_found)`